src/middleware/secure-headers/secure-headers.ts

```diff
@@ -255,6 +272,36 @@ function getCSPDirectives(
       ]
 }
 
+function getPermissionsPolicyDirectives(policy: PermissionsPolicyOptions): string {
```

Comment:
- Reused camelToKebab(directive) by storing it in a kebabDirective variable, reducing redundant function calls.
- Simplified array handling by removing unnecessary checks for specific values like 'self' and 'src'.
- Optimized empty array handling by using a concise conditional, improving code readability and efficiency.

```ts
function getPermissionsPolicyDirectives(policy: PermissionsPolicyOptions): string {
  return Object.entries(policy)
    .map(([directive, value]) => {
      const kebabDirective = camelToKebab(directive)

      if (typeof value === 'boolean') {
        return `${kebabDirective}=${value ? '()' : 'none'}`
      }

      if (Array.isArray(value)) {
        const allowlist = value.length ? value.join(' ') : ''
        return `${kebabDirective}=(${allowlist})`
      }

      return ''
    })
    .filter(Boolean)
    .join(', ')
}

```